client/recoverSecret.rs

```diff
@@ -0,0 +1,136 @@
+                    // - Cut from current index of result to the first letter of word
+                    // - Put it just after the precedent letter index of word
+                    if tmp_index < first_word_letter {
+                        let mut cutted = &result[tmp_index as usize..first_word_letter as usize].to_string();
+                        let new_resutlt = result[..tmp_index as usize].to_string() + &result[first_word_letter as usize..prec_plus as usize].to_string() + &*cutted.to_string() + &result[prec_plus as usize..].to_string();
+                        result = new_resutlt;
+                        // println!("result 1 : {}", result);
+                    }
+                    else{
+                        let mut cutted = &result[tmp_index as usize..prec_letter as usize].to_string();
+                        let new_resutlt = result[0..tmp_index as usize].to_string() + &result[prec_letter as usize..].to_string() + &*cutted.to_string();
+                        result = new_resutlt;
+                        // println!("result 2: {}", result);
+                    }
+                }
+                prec_letter = i as i32;
+            }
+            // println!("\nNew result: {}\n", result);
+        }
+        status = validateRecoverSecretSentence(result.clone().to_string(), tuple_sizes.clone(), letters.clone())
+    }
+
+    let mut output: RecoverSecretOutput = RecoverSecretOutput{
+        secret_sentence: result
+    };
+    return output;
+}
+
+// Get the index of letter parameter in word parameter as parameters
+pub fn getValueIndex(word: String, letter: String) -> i32 {
+    for j in 0..word.len(){
+        if word.chars().nth(j).unwrap() == letter.chars().nth(0).unwrap(){
+            return j as i32;
+        }
+    }
+    return -1;
+}
+
+// Verify if a word or sentence respecting the tuple_sizes and letters parameters rules
+pub fn validateRecoverSecretSentence(test_word: String, tuple_sizes: Vec<usize>, letters: String) -> bool {
+
+    // Validate a sentence test_word with letters and tuple_sizes
+    for i in 0..tuple_sizes.len(){
+        let mut startInterval = 0;
+        let mut endInterval = 0;
+        for j in 0..i{
+            startInterval = startInterval + tuple_sizes[j];
+            endInterval = endInterval + tuple_sizes[j];
+        }
+        endInterval = endInterval + tuple_sizes[i];
+
+        let mut word = &letters[startInterval..endInterval];
+        // println!("{}", patternWord(word, test_word));
+        let status = patternWord(word.to_string(), test_word.to_string());
+        if (!status){
+            return false;
+        }
+    }
+    return true;
+}
+
+// Verify if a Vector of i32 is sorted
+fn is_sorted<T>(data: Vec<i32>) -> bool
+    where
+        T: Ord,
+{
+    data.windows(2).all(|w| w[0] <= w[1])
+}
+
+// Verify if word parameter repects pattern variable
+fn patternWord(pattern:String, word: String) -> bool {
```

Comment:
same for it put it at the top in generaly we can let it at the bottom because there are private just for our main function but right there they need to be understand before we read the main function :)

Reply:
ok